In [1]:
from __future__ import print_function, division
%matplotlib inline
import statsmodels.formula.api as smf  
import matplotlib.pyplot as plt
import matplotlib
import pandas as pd
import numpy as np
import geopandas as gpd
import os
import sys
from scipy import stats

#### Likelihood ratio test

Null hypothsis: the complex model is not better than the simple one.

In [5]:
print ("LR compare #1 and #2: ", -2 * ((-160.985) - (-60.4931)))

LR compare #1 and #2:  200.98380000000003


#### The critical chi squared value for 1 degree of freedom is 3.84 < 200.984, so we can reject the Null hypothsis. We think the model 2 is a better choice here.

In [4]:
print ("LR compare #2 and #3: ", -2 * ((-60.5266) - (-60.4931)))

LR compare #2 and #3:  0.06700000000000728


In [3]:
print ("LR compare #3 and #4: ", -2 * ((-60.7518) - (-60.5266)))

LR compare #3 and #4:  0.4504000000000019


#### The critical chi squared value for 1 degree of freedom is 3.84 >0.45, so we can not reject the Null hypothsis. We think the complex model is not a better choice here.